packages/js/src/plugins/auctionHouseModule/operations/withdrawFromBuyerAccount.ts

```diff
@@ -0,0 +1,251 @@
+ */
+export const withdrawOperationHandler: OperationHandler<WithdrawOperation> = {
+  handle: async (operation: WithdrawOperation, metaplex: Metaplex) =>
+    withdrawFromBuyerAccountBuilder(operation.input, metaplex).sendAndConfirm(
+      metaplex,
+      operation.input.confirmOptions
+    ),
+};
+
+// -----------------
+// Builder
+// -----------------
+
+/**
+ * @group Transaction Builders
+ * @category Inputs
+ */
+export type WithdrawFromBuyerAccountBuilderParams = Omit<WithdrawInput, 'confirmOptions'> & {
+  instructionKey?: string;
+};
+
+/**
+ * @group Transaction Builders
+ * @category Contexts
+ */
+export type WithdrawBuilderContext = Omit<WithdrawOutput, 'response'>;
+
+/**
+ * Withdraws funds from the user's buyer escrow account to the given auction house.
+ *
+ * ```ts
+ * const transactionBuilder = metaplex
+ *   .auctionHouse()
+ *   .builders()
+ *   .withdrawFromBuyerAccountBuilder({ auctionHouse, buyer, amount });
+ * ```
+ *
+ * @group Transaction Builders
+ * @category Constructors
+ */
+export const withdrawFromBuyerAccountBuilder = (
+  metaplex: Metaplex,
+  params: WithdrawFromBuyerAccountBuilderParams
+): TransactionBuilder<WithdrawBuilderContext> => {
+  const {
+    auctionHouse,
+    auctioneerAuthority,
+    amount,
+    payer = metaplex.identity(),
+  } = params;
+
+  if (auctionHouse.hasAuctioneer && !params.auctioneerAuthority) {
+    throw new AuctioneerAuthorityRequiredError();
+  }
+
+  const amountBasisPoint = amount.basisPoints;
+  const buyer = params.buyer ?? (metaplex.identity() as Signer);
+  const authority = params.authority ?? auctionHouse.authorityAddress;
+
+  if (!isSigner(buyer) && !isSigner(authority)) {
+    throw new WithdrawRequiresSignerError();
+  }
+
+  const escrowPayment = findAuctionHouseBuyerEscrowPda(
+    auctionHouse.address,
+    toPublicKey(buyer)
+  );
+
+  // Accounts,
+  const accounts: WithdrawInstructionAccounts = {
+    wallet: toPublicKey(buyer),
+    receiptAccount: toPublicKey(buyer),
+    escrowPaymentAccount: escrowPayment,
+    treasuryMint: auctionHouse.treasuryMint.address,
+    authority: toPublicKey(authority),
+    auctionHouse: auctionHouse.address,
+    auctionHouseFeeAccount: auctionHouse.feeAccountAddress,
+  };
+
+  // Args.
+  const args = {
+    escrowPaymentBump: escrowPayment.bump,
+    amount: amountBasisPoint,
+  };
+
+  // Withdraw Instruction.
+  let withdrawInstruction = createWithdrawInstruction(accounts, args);
+  if (auctioneerAuthority) {
+    const ahAuctioneerPda = findAuctioneerPda(
+      auctionHouse.address,
+      auctioneerAuthority.publicKey
+    );
+
+    const accountsWithAuctioneer = {
+      ...accounts,
+      auctioneerAuthority: auctioneerAuthority.publicKey,
+      ahAuctioneerPda,
+    };
+
+    withdrawInstruction = createAuctioneerWithdrawInstruction(
+      { ...accountsWithAuctioneer },
+      args
+    );
+  }
+
+  // Signers.
+  const withdrawSigners = [buyer, authority, params.auctioneerAuthority].filter(
```

Comment:
Try another test, create auctionHouse with different authority and try to provide this authority as a signer to this operation.

Reply:
You need this thing similarly to createListing to correctly apply signer. 

```
// Update the account to be a signer since it's not covered properly by MPL due to its dynamic nature.
  const signerKeyIndex = sellInstruction.keys.findIndex((key) =>
    key.pubkey.equals(signer.publicKey)
  );
  sellInstruction.keys[signerKeyIndex].isSigner = true;
```

Reply:
Currently, it works because fee payer equals authority.